src/puppetlabs/trapperkeeper/internal.clj

```diff
@@ -44,6 +47,25 @@
    (every? keyword? (keys service-graph))
    (every? (some-fn ifn? service-graph?) (vals service-graph))))
 
+(defn inc-restart-counter!
+  "Increments the counter in a restart-file for purposes of supporting HUP behavior"
+  [restart-file-path]
+  (try
+    (if (fs/exists? restart-file-path)
+      (if (and (fs/writeable? restart-file-path) (fs/readable? restart-file-path))
+        (spit restart-file-path (inc (Long/parseLong (slurp restart-file-path))))
+        (log/error "Restart File is not readable and/or writeable"))
```

Comment:
May be useful to include the file we can't read/write here, as we know what it is.